sdb/commands/pretty_print.py

```diff
@@ -30,28 +31,30 @@ class PrettyPrint(sdb.Command):
     def call(self, objs: Iterable[drgn.Object]) -> None:  # type: ignore
         baked = [(self.prog.type(type_), class_)
                  for type_, class_ in sdb.PrettyPrinter.all_printers.items()]
-        has_input = False
-        for i in objs:
-            has_input = True
-
+        handlingClass = None
+        inputType = None
+        firstObj = next(iter(objs), None)
+        if firstObj:
+            inputType = firstObj.type_
             try:
                 for type_, class_ in baked:
-                    if i.type_ == type_ and hasattr(class_, "pretty_print"):
-                        class_(self.prog).pretty_print([i])
+                    if inputType == type_ and hasattr(class_, "pretty_print"):
+                        handlingClass = class_
                         raise StopIteration
             except StopIteration:
-                continue
-
-            # error
-            raise TypeError(
-                'command "{}" does not handle input of type {}'.format(
-                    self.names, i.type_))
-        # If we got no input and we're the last thing in the pipeline, we're
-        # probably the first thing in the pipeline. Print out the available
-        # pretty-printers.
-        if not has_input and self.islast:
+                pass
```

Comment:
Out of curiosity - why can't we just use `break` here?

Reply:
I think this evolved from a previous iteration + my unfamiliarity with python.  Thanks for pointing this out, I've simplified it now.